### PR TITLE
Machinery EMP Resistance (Wild Singulo Fixes)

### PR DESCRIPTION
**Pull request by Azrun**
[balance][bug - major][input wanted]
## About the PR 
* Singularities will no longer attempt to eat things that do not exist. [Bug]
* Limits Camera's to only queue only recalculation when multiple EMPs have happened in the 90 second disable window. [Performance]
  * This does make it so you can not add additional delay until the cameras comes back online. [Balance]
* Limits APC battery drain from EMP to once every 90 seconds. [Performance] [Balance]
   * Battery drain from EMP can be mitigated by turning off APC due to implementation [Balance]
* Limits EMP on standard machinery to only procing overlay and power drain to once per second.

This does end up being an "unintentional" EMP Grenades, Wave Gun, EMP* nerf as there will be a limit to how often those items can impact /obj/machinery. 

----

Addresses a number of runtimes that occur due to spawned tasks having their object disposed of.
Targets performance improvements when multiple EMPs go off to limit the number of new tasks spawned.

## Why's this needed? 
Runtimes are bad.
Resolve unusual behavior from multiple EMP's going off.  Like when the station is covered in singularities...

```
(u)Azrun:
(+)Machinery, cameras, and APCs are more EMP resistant.  Cameras will come back online after being EMPed after 90 seconds, duration will no longer be extended with additional EMPs.
```

### CODE REVIEW

**Comment by flrsh** [comment]
Can you rephrase `Will no longer be able to re-EMPs to reset duration that camera is disabled until it comes back online.` so that it's more clear?

File: _std/machinery.dm [hunk 1]
```diff
@@ -18,11 +18,13 @@
 #define TOTAL 4	//for total power used only
 
 // bitflags for machine stat variable
-#define BROKEN 1		// machine non-functional
-#define NOPOWER 2		// no available power
-#define POWEROFF 4		// machine shut down, but may still draw a trace amount
-#define MAINT 8			// under maintainance
-#define HIGHLOAD 16		// using a lot of power
+#define BROKEN    (1<<0)		// machine non-functional
```
Comment (ZeWaka) [inline _std/machinery.dm:21]: Can you put these in the DMdoc style

Comment (Azrun) [reply _std/machinery.dm]: Sure! 
I find code diving faster and less error prone so I never spent much looking at the Generated Docs, so I guess I'll try to stop being part of the problem. 😉

File: _std/machinery.dm [hunk 2]
```diff
@@ -12,17 +12,19 @@
...
 
 //apc stuff
-#define EQUIP 1
-#define LIGHT 2
-#define ENVIRON 3
-#define TOTAL 4	//for total power used only
+#define EQUIP 1 	//! Power Channel: Equipment
+#define LIGHT 2 	//! Power Channel: Lighting
+#define ENVIRON 3 //! Power Channel: Enviroment
+#define TOTAL 4		//! For total power used only
```
Comment (ZeWaka) [inline _std/machinery.dm:18]: those won't work, you need to do the `///` style

Comment (Azrun) [reply _std/machinery.dm:18]: Huh, it worked locally using the released executable. 😕  Must only partially work with that syntax... I'll poke it again during my lunch break.

Comment (Azrun) [reply _std/machinery.dm:18]: Line numbers are different... otherwise appear identical.
https://github.com/Azrun/goonstation/commit/9964153a00089275604351f005e3ef08fe9890bf

What am I missing?

Comment (ZeWaka) [reply _std/machinery.dm:18]: Maybe @SpaceManiac might know more? It never worked for me when I used it in live docs.

Comment (SpaceManiac) [reply _std/machinery.dm:18]: what works and what doesn't unfortunately differs between `#define` and other declarations such as vars

Comment (Azrun) [reply _std/machinery.dm:18]: Thanks!

**Comment by Azrun** [comment]
@ZeWaka you want me to change it for consistency or is this pending anything else?

**Review by ZeWaka — APPROVED** [review]
